File: checkinpromobile/android/app/src/main/kotlin/vn/com/unit/checkinpromobile/PrinterInfor.kt
package vn.com.unit.checkinpromobile

class PrinterInfor {
    var type = ""

    var ipAddress = ""

    var model = ""

    var isConnect = false

    constructor(type: String, ipAddress: String, model: String, isConnect: Boolean) {
        this.type = type
        this.ipAddress = ipAddress
        this.model = model
        this.isConnect = isConnect
    }

    constructor()
}

File: checkinpromobile/lib/src/model/PrinterInfor.dart
import 'package:json_annotation/json_annotation.dart';

part 'PrinterInfor.g.dart';

@JsonSerializable()
class PrinterInfor {
  @JsonKey(name: 'actions')
  String actions;

  @JsonKey(name: 'ipAddress')
  String ipAddress;

  @JsonKey(name: 'model')
  String model;

  @JsonKey(name: 'isConnect')
  bool isConnect;

  PrinterInfor._();

  PrinterInfor(this.actions, this.ipAddress, this.model, this.isConnect);

  factory PrinterInfor.fromJson(Map<String, dynamic> json) =>
      _$PrinterInforFromJson(json);

  Map<String, dynamic> toJson() => _$PrinterInforToJson(this);
}
